lms/djangoapps/discussion_api/api.py

```diff
@@ -177,103 +185,136 @@ def get_course(request, course_key):
-        "non_courseware_topics": non_courseware_topics,
-    }
 
+def get_non_courseware_topics(request, course_key, course, topic_ids):
+    """
+    Returns a list of topic trees that are not linked to courseware.
+
+    Parameters:
+
+        request: The django request objects used for build_absolute_uri.
+        course_key: The key of the course to get discussion threads for.
+        course: The course for which topics are requested.
+        topic_ids: A list of topic IDs for which details are requested.
+            This is optional. If None then all course topics are returned.
+
+    Returns:
+        A list of non-courseware topics and a list of existing topics among
+        topic_ids.
 
-def get_course_topic(request, course_key, topic_id):
     """
-    Returns details of discussion topic for the given topic_id.
+    non_courseware_topics = []
+    existing_topic_ids = []
+    for name, entry in sorted(course.discussion_topics.items(), key=lambda item: item[1].get("sort_key", item[0])):
+        if not topic_ids or entry['id'] in topic_ids:
+            discussion_topic = DiscussionTopic(
+                entry["id"], name, get_thread_list_url(request, course_key, [entry["id"]]),
+            )
+            non_courseware_topics.append(discussion_topic.__dict__)
+
+            if topic_ids and entry["id"] in topic_ids:
+                existing_topic_ids.append(entry["id"])
+
+    return non_courseware_topics, existing_topic_ids
+
+
+def get_course_topics(request, course_key, topic_ids=None):
+    """
+    Returns the course topic listing for the given course and user; filtered
+    by 'topic_ids' list if given.
 
     Parameters:
 
-        request: The django request object
-        course_key: course_key: The key of the course to get discussion threads for
-        topic_id: Topic ID string to get discussion
+        course_key: The key of the course to get topics for
+        user: The requesting user, for access control
+        topic_ids: A list of topic IDs for which topic details are requested
 
     Returns:
 
-        Discussion module for given topic id.
+        A course topic listing dictionary; see discussion_api.views.CourseTopicViews
+        for more detail.
 
     Raises:
-
-        DiscussionNotFoundError: If discussion module does not exist for given topic
-        id or requesting user does not have access to requested discussion module.
+        DiscussionNotFoundError: If topic/s not found for given topic_ids.
     """
     course = _get_course(course_key, request.user)
-    discussion_module = get_discussion_module(course, request.user, topic_id)
+
+    courseware_topics, existing_courseware_topic_ids = get_courseware_topics(request, course_key, course, topic_ids)
+    non_courseware_topics, existing_non_courseware_topic_ids = get_non_courseware_topics(
+        request, course_key, course, topic_ids
+    )
+
+    if topic_ids:
+        not_found_topic_ids = list(set(topic_ids) - (
```

Comment:
Shouldn't need to convert this back to a list.  Let not_found_topic_ids remain a set.